README.md

```diff
@@ -0,0 +1,111 @@
+# stream-json-rpc
+Call remote process methods easily, using the transport that you need.
+
+## Usage
+Example using node-ipc
+```typescript
+// The transport layer must implement the `stream.Duplex` interface
+// Here are the ones for `node-ipc`
+import { Duplex } from 'stream';
+
+class ServerDuplex extends Duplex {
+  ipc: NodeIPC.Server; // instance of nodeipc.server
+  socket: NodeIPC.Socket; // destination socket
+
+  constructor(ipc: NodeIPC.Server, socket: NodeIPC.Socket) {
+    super();
+    this.ipc = ipc;
+    this.socket = socket;
+
+    ipc.on('data', data => {
+      this.push(data);
+    });
+  }
+
+  // tslint:disable-next-line
+  _write(chunk: any, _encoding: any, callback: any) {
+    this.ipc.emit(this.socket, chunk);
+    callback();
+  }
+
+  // tslint:disable-next-line
+  _read(_size: any) {}
+}
+
+class ClientDuplex extends Duplex {
+  ipc: NodeIPC.Client; // ipc socket opened between current process and the server
+
+  constructor(ipc: NodeIPC.Client) {
+    super();
+    this.ipc = ipc;
+
+    ipc.on('data', data => {
+      this.push(data);
+    });
+  }
+
+  // tslint:disable-next-line
+  _write(chunk: any, _encoding: any, callback: any) {
+    this.ipc.emit(chunk);
+    callback();
+  }
+
+  // tslint:disable-next-line
+  _read(_size: any) {}
+}
+```
+
+And the actual implementation would look like:
+##### Process 1
+```typescript
+import { rpcchannel } from 'stream-json-rpc';
+
+// This process acts as node-ipc server
+// But it actually being a "server" is not really relevant,
+// as any process can call any other process if they can directly communicate.
+
+const ipc = getIPC(); // Instance of NodeIPC.Server
+const sockets = new Map(); // List of clients
+const channel = rpcchannel(); // Create the channel on the server side
+
+// At first connection, store the client socket
+const firstConnection = (data: Buffer, socket: any) => {
+  const id = data.toString('utf-8');
+  sockets.set(id, socket);
+  channel.setLink(id, new ServerDuplex(ipc, socket));
+  ipc.off('data', firstConnection);
+};
+ipc.on('data', firstConnection);
+
+// Register handlers
+channel.addRequestHandler('inc', ({ value }: any) => {
```

Comment:
I don't think having 2 handlers for a given method is a good pattern. I'd rename this `setRequestHandler `

Reply:
BTW I think the implementation of 'json-rpc-peer' to add handler is not nice and could evolve to add this method `setRequestHandler` et `setNotifyHandler`